### PR TITLE
[SPARK-31253][SQL][FOLLOW-UP] simplify the code of calculating size metrics of AQE shuffle 

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

A followup of https://github.com/apache/spark/pull/28175:
1. use mutable collection to store the driver metrics
2. don't send size metrics if there is no map stats, as UI will display size as 0 if there is no data
3. calculate partition data size separately, to make the code easier to read.

### Why are the changes needed?

code simplification

### Does this PR introduce any user-facing change?

no

### How was this patch tested?

existing tests

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @JkSelf @maryannxue

File: sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala [hunk 1]
```diff
@@ -97,48 +97,42 @@ case class CustomShuffleReaderExec private(
...
-        }
-
-        // Set sum value to "partitionDataSize" metric.
-        partitionMetrics.set(sum)
-      }
+    partitionDataSizes.foreach { dataSizes =>
+      val partitionDataSizeMetrics = metrics("partitionDataSize")
+      driverAccumUpdates ++= dataSizes.map(partitionDataSizeMetrics.id -> _)
+      // Set sum value to "partitionDataSize" metric.
+      partitionDataSizeMetrics.set(dataSizes.sum)
```
Comment (JkSelf) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala:135]: why use the `dataSizes's sum` not `partitionDataSizes's sum` ?

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala:135]: `partitionDataSizes.foreach { dataSizes =>` so `dateSizes = partitionDataSizes.get`

**Review by maryannxue — APPROVED** [review]
LGTM.

**Comment by gatorsmile** [comment]
Thanks! Merged to master/3.0

**Comment by jiangxb1987** [comment]
Merged to master only, please manually backport this to 3.0. Thanks!

**Comment by cloud-fan** [comment]
We haven't decided to backport the AQE SQL metrics yet, so this PR can't be backported for now.

**Pull request closed without merging**